### PR TITLE
fix: Update Authorization Prompt URL to Use URL Anchor Instead of Query Parameters (WPB-1343)

### PR DESCRIPTION
**Pull request by thisisamir98**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-1343" title="WPB-1343" target="_blank">BugWPB-1343</a>  [Web] [OAuth] OAuth parameters are sent to webapp server for Authorization Prompt
  </td></table>
  <br />
 

## Description

This PR modifies the Authorization Prompt URL to add URL anchors to OAuth URLs. By doing so, we ensure that potentially sensitive data, which is solely managed by client-side, is no longer sent to the backend server. 

## Screenshots/Screencast (for UI changes)

https://github.com/wireapp/wire-webapp/assets/63250054/7f302922-14d7-4899-b00a-860fa81387f8

## Checklist

- [X] PR has been self reviewed by the author;
- [X] Hard-to-understand areas of the code have been commented;
- [X] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/auth/page/Login.tsx [hunk 1]
```diff
@@ -182,7 +182,8 @@ const LoginComponent = ({
       await doInitializeClient(ClientType.PERMANENT, undefined, undefined, entropyData);
 
       if (isOauth) {
-        return navigate(ROUTE.AUTHORIZE);
+        console.info('bardia route', {route: `${ROUTE.AUTHORIZE}/${window.location.hash}`});
```
Comment (PatrykBuniX) [inline src/script/auth/page/Login.tsx]: 👀

Comment (thisisamir98) [reply src/script/auth/page/Login.tsx]: thanks for the catch: https://github.com/wireapp/wire-webapp/pull/15844/commits/158a9b9be132be161ba1c13e7b125d5224a9457e

**Comment by tlebon** [comment]
not working when signing out of account

**Comment by thisisamir98** [comment]
@tlebon it is handled now in https://github.com/wireapp/wire-webapp/pull/15844/commits/102786c82ffc07e1d9669dcf10c1dfbe2596ba0e

File: src/script/auth/page/Root.tsx [hunk 2]
```diff
@@ -107,7 +107,20 @@ const RootComponent: FC<RootProps & ConnectedProps & DispatchProps> = ({
   };
 
   const isAuthenticatedCheck = (page: any): any => (page ? (isAuthenticated ? page : navigate('/auth')) : null);
-  const isOAuthCheck = (page: any): any => (page ? isAuthenticated ? page : <Navigate to={ROUTE.LOGIN} /> : null);
+  const isOAuthCheck = (page: any): any => {
```
Comment (przemvs) [inline src/script/auth/page/Root.tsx]: We don't know this type?

Comment (thisisamir98) [reply src/script/auth/page/Root.tsx]: don't know, its been like this forever

Comment (thisisamir98) [reply src/script/auth/page/Root.tsx]: https://github.com/wireapp/wire-webapp/pull/15844/commits/a1096597c683ab65ebdccfb3d8319a3827cba5af

**Review by tlebon — APPROVED** [review]
looks good

**Pull request merged**